example/craco.config.js

```diff
@@ -0,0 +1,9 @@
+const path = require("path");
+
+module.exports = {
+    webpack: {
+        alias: {
+            "react": path.join(path.resolve(__dirname, "../node_modules/react"))
+        }
+    }
+}
```

Comment:
This is a gnarly one. The short story is that you cannot run two separate installed instances of React at once. When developing locally with `react-showtime` linked and running it in development mode, it has its own version of `react` installed to support testing. This causes a conflict with the `react` installed by `example`, even though they're both the same version.

Previously this was addressed by linking to the parent React version using `link` in `package.json`, but that won't work for Netlify builds. This config extends the CRA Webpack config to alias any `react` imports in the `example` project to the `react` installed by `react-showtime`, in the parent directory. Since we only invoke `craco` when running `yarn start`, this alias is only used when running a local development server and doesn't impact builds.

Reply:
It looks like Scott used [create-react-hook](https://github.com/Hermanya/create-react-hook) to initialize this project. The template it produces includes the concept of a [library in the project root](https://github.com/Hermanya/create-react-hook/tree/master/template/default) with an [example site](https://github.com/Hermanya/create-react-hook/tree/master/template/default/example) alongside it. It also appears to handle [building a static bundle of the example site](https://github.com/Hermanya/create-react-hook/blob/5ca757eb55e040040c84252d16b28d397250586d/template/default/package.json#L21-L22) for deployment (but in their example, it aims to target GitHub Pages).

Despite the fact that it seems like our configuration works, we're deviating a bit from the base, bringing in `craco` for one purpose, and it still seems possible that their version has resolved issues that ours hasn't encountered yet. That makes me think that it may be worth considering if we should continue down the existing path, or attempt to align ourselves with what `create-react-hook` is doing.

Reply:
I had completely missed the bit about `create-react-hook` being used to generate this project in the main README. Based on the `example` README, I had assumed that the example site was a stripped-down `create-react-app` project and the rest was put together by hand. That adds a ton of context and clarity for me.

I've reworked things significantly to better align with the base setup for `create-react-hook`:

- `craco` has been removed.
- I've adapted the `yarn run deploy` command to work for Netlify instead of GitHub Pages, and the build is now run from the repository root rather than the `example` directory. The `gh-pages` package has also been removed.
- I've also restored the `react` `link` dependency, since it makes much more sense now that we're building from the repo root.